frontend/src/libs/types/form.type.ts

```diff
@@ -1,10 +1,31 @@
-import { type FieldPath, type FieldValues } from 'react-hook-form';
+import {
+  type ErrorOption,
+  type FieldPath,
+  type FieldValues,
+} from 'react-hook-form';
+
+import { type HttpMessage } from '~/libs/enums/enums.js';
+
+import { type ValueOf } from './types.js';
+
+type KnownErrorMessages = ValueOf<typeof HttpMessage>;
+
+type ErrorDescriptor =
+  | {
+      errorMessage: KnownErrorMessages;
+      error: ErrorOption;
+      options?: {
+        shouldFocus: boolean;
+      };
+    }
+  | KnownErrorMessages;
```

Comment:
Can we have 1 type for errors? Otherwise we'll have to verify everywhere is it object or a string

Reply:
So, we can't?